File: ACM-ICPC/LeetCode/P704.cpp
// 704. 二分查找
class Solution
{
public:
    int search(vector<int> &nums, int target)
    {
        int ll = 0, rr = nums.size() - 1;
        while (ll <= rr)
        {
            int mid = (ll + rr) / 2;
            if (nums.at(mid) == target)
            {
                return mid;
            }
            else if (nums.at(mid) < target)
            {
                ll = mid + 1;
            }
            else if (nums.at(mid) > target)
            {
                rr = mid - 1;
            }
        }
        return -1;
    }
};

File: ACM-ICPC/LeetCode/P704.go
// 704. 二分查找
func search(nums []int, target int) int {
	l, r := 0, len(nums)
	for l < r {
		m := (l + r) / 2
		if nums[m] < target {
			l = m + 1
		} else {
			r = m
		}
	}
	if l >= len(nums) || nums[l] != target {
		return -1
	}
	return l
}
